harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -184,6 +195,33 @@ def download(self, path: Optional[str] = None) -> str:
 
         return str(local_ckpt_dir)
 
+    def _download_via_master(self, local_ckpt_dir: pathlib.Path) -> None:
+        local_ckpt_dir.mkdir(parents=True, exist_ok=True)
+        if os.name == "nt":
+            fext = "zip"
+            mime_type = "application/zip"
+        else:
+            fext = "tgz"
+            mime_type = "application/gzip"
+        resp = self._session.get(f"/checkpoints/{self.uuid}", headers={"Accept": mime_type})
```

Comment:
I think you can set `stream=True` here... otherwise you'll be at risk of OOM issues on a regular basis.

Reply:
Sure, thought about that but didn't dig enough.